exporter/datadogexporter/factory_test.go

```diff
@@ -389,10 +389,91 @@ func TestCreateAPIMetricsExporter(t *testing.T) {
 	assert.NotNil(t, exp)
 }
 
+func TestCreateAPIExporterFailOnInvalidKey_Zorkian(t *testing.T) {
+	server := testutil.DatadogServerMock(testutil.ValidateAPIKeyEndpointInvalid)
+	defer server.Close()
+
+	err := enableZorkianMetricExport()
+	require.NoError(t, err)
```

Comment:
Should we add a `defer <function that restores feature gate previous value>()` to tests? I am a bit worried that, since we are modifying global state, this will accidentally leak into other tests and we would be using a different metric export from the one we expect on a given test.

(Another option would be to pass the feature gate registry explicitly, but I think that would need more pervasive changes, so defer sounds like a reasonable compromise)